aws-android-sdk-mobile-client/src/main/java/com/amazonaws/mobile/client/AWSMobileClient.java

```diff
@@ -1611,6 +1611,9 @@ public Void run() throws Exception {
                     userpoolLL.globalSignOut(globalSignOutRequest);
                 }
                 if (signOutOptions.isInvalidateTokens()) {
+                    if (userpool != null) {
+                        userpool.getCurrentUser().revokeTokens();
+                    }
```

Comment:
If the revokeTokens call fails for any reason (poor internet connection, server error, etc.), will the custom tabs browser still be launched to attempt to clear the cookies?  I think we should.

Also, if revokeTokens fails, does the failure bubble up to the signout callback so the user knows that the tokens were not revoked?

Reply:
Yes, if revokeTokens fails signout should continue. Revocation is best effort so a failure will be logged but the failure exception will be not be bubbled up.